class-udm-updater.php

```diff
@@ -95,6 +100,34 @@ public function __construct($mothership, $muid, $relative_plugin_file, $options
 		if (version_compare($wp_version, '5.5', '<')) {
 			add_filter('auto_update_plugin', array($this, 'auto_update_plugin'), 20, 2);
 		}
+
+		add_action('admin_init', array($this, 'disconnect_cloned_site'));
+	}
+
+	/**
+	 * Disconnect a cloned site by unsetting the `email` option and resetting PUC update cache and at the same time adding `site_host_path` as a plugin option for a cloned site checking purpose in future
+	 */
+	public function disconnect_cloned_site() {
+
+		global $pagenow; 
+
+		if (defined('DOING_AJAX') && DOING_AJAX || 'plugins.php' !== $pagenow) return;
+
+		$udm_options = $this->get_option($this->option_name);
+		$site_url = parse_url(site_url());
```

Comment:
@DavidAnderson684 That's really a nice catch, thanks for spotting that one. It was confusing me at first because I kept thinking that a sub-site would have its own unique `Site ID` which was different from the network domain (I overlooked the `sitemeta` part), but after performing another test on a multisite, I find that no matter how many sub-sites there are, they will just use the same site id from the sitemeta table. Also, I've changed the `home_url()` to `network_site_url()` to all parts that were using it in this library. However, replacing the function will make some websites that use this library detected as a cloned site due to difference of URLs  and when the users reconnect they will get the `This site was previously using a licence also used by another WordPress install with a different URL (which most likely originates from one of the sites being created by duplicating the other) - this has resulted in the licence being disconnected from this site` message, is that okay with you?

Reply:
@andtown Ah yes, I hadn't thought of that problem. Do you think if `is_multisite()` is true and if the two values differ we could also send the value of `home_url()`, and do the update silently on the server (i.e. don't complain about duplication) if `home_url()` is matching the old URL?

Reply:
@DavidAnderson684 I've looked over the code on both libraries (client and server) and yes I think that seems to be the only way (and tend to be the best) to workaround the problem as I couldn't see any other way. So, I propose the below changes on both libraries:

`Updraft_Manager_Plugin::get_plugin_info() - updraftmanager-plugin.php` **(Server)**

![image](https://user-images.githubusercontent.com/400912/112753927-ef5bbb00-9003-11eb-8d79-ca2932fff7c0.png)

as you can see in the image above, the `hu` key name stands for `home url`, I don't use a new/different variable name and just stick with the $_POST variable to avoid changing the function signature. The `!empty($_POST['hu'])` will make sure compatibility with older versions of client-side library. The `$entitlement['url_normalised'] !== Updraft_Manager::normalise_url(base64_decode($_POST['hu']))` will make sure that, when `home_url()` matches the old URL, `x-spm-duplicate-of` header won't get issued, note that the old URL in the entitlements table gets updated to the new one before the `if` check takes place.

`Updraft_Manager_Updater::updater_queryargs_plugin() - class-udm-updater.php` **(Client)**

![image](https://user-images.githubusercontent.com/400912/112754072-abb58100-9004-11eb-979f-54c0a02c728a.png)

`Updraft_Manager_Updater::connect() - class-udm-updater.php` **(Client)**

![image](https://user-images.githubusercontent.com/400912/112754138-fdf6a200-9004-11eb-937b-e512d0816061.png)

I've given it a quick test and all work as expected. If you want those code to be implemented, I will issue a new MR for the server-side library. If you have other thoughts, please let me know.

Reply:
@DavidAnderson684 I'm just sending you a new email notification in case you missed the previous one.

Reply:
@andtown OK, yes, let's do that. But call it `home_url` instead of `hu` - the unnecessary abbreviation in the first release many years ago was a bad idea (saving a few bytes doesn't do anything useful, it just makes the code harder to read).

Reply:
@DavidAnderson684 I've sent you an MR that deals with changes on the server-side library. Also, new changes in this PR as discussed.